tests/baselines/reference/forInGenericPrimitive.errors.txt

```diff
@@ -0,0 +1,31 @@
+tests/cases/compiler/forInGenericPrimitive.ts(15,21): error TS2407: The right-hand side of a 'for...in' statement must be of type 'any', an object type or a type parameter, but here has type 'B'.
+tests/cases/compiler/forInGenericPrimitive.ts(16,21): error TS2407: The right-hand side of a 'for...in' statement must be of type 'any', an object type or a type parameter, but here has type 'C'.
+tests/cases/compiler/forInGenericPrimitive.ts(17,21): error TS2407: The right-hand side of a 'for...in' statement must be of type 'any', an object type or a type parameter, but here has type 'D'.
+
+
+==== tests/cases/compiler/forInGenericPrimitive.ts (3 errors) ====
+    function f<
+    A,
+    B extends { a: number } | string,
+    C extends string | void,
+    D extends number | { s: string } | null | string,
+    E extends { x: number },
+    >(
+        a: A,
+        b: B,
+        c: C,
+        d: D,
+        e: E,
+    ) {
+        for (const _ in a) { }

```

Comment:
Unconstrained generics don't issue an error because `unknown` (the constraint) isn't assignable to any of the primitive kinds - this keeps the breakage of this change down by still permitting for-in over an unconstrained parameter. We could swap to comparability of this was undesirable.

Reply:
It's a good point. Typically, the right-hand side of a `for...in` statement must typically be of type 'any', an object type, or a type parameter, however in the case of unconstrained generics, this seems to allow for 'unknown' (the constraint of unconstrained generics), which isn't assignable to any of the primitive kinds.

This design decision, in which unconstrained generics don't issue an error, permits `for...in` loops over unconstrained parameters, which can be useful but may have consequences.

It might be beneficial to further discuss the implications of this decision and whether it aligns with the intended use of unconstrained generics in the codebase.